AWS Migration

toadzky opened the pull request:
Switches to gremlin for the backend, adds cloudformation, and some github actions.

toadzky committed:
WIP - first stab at github actions

toadzky committed:
Adds distribution management block

toadzky committed:
adds patch version and test docker command

toadzky committed:
adds release workflow for master, builds docker, doesn't deploy

toadzky committed:
WIP - convert to gremlin

toadzky committed:
Switch to gremlin for neptune compatibility

toadzky committed:
Updates readme for neptune/aws

toadzky committed:
Adds cloudformation stuff, makes a release workflow, updates readme images

toadzky committed:
Fixes depednecy issues

toadzky committed:
Adds aws signage, plays with dependencies some more

toadzky committed:
Clean up some small things and get it building

ndrwksr commented:
I am clueless about anything here but the Kotlin... That said, the Kotlin looks good to me!

toadzky committed:
Fixes all the cloudformation and startup issues. Deploying is successful

toadzky committed:
small readme update

toadzky committed:
Fixes local runs, fixes gremlin vertex query calls

yonglid reviewed (changes requested):
Already left Slack comments but just so we have it recorded here (and it doesn't get lost) 
- updated changes made by David (custom made data classes like DeviceId need to be unwrapped ref by value when referenced by graph api) 
- talked about adjusting gremlin settings for vertex indices so they aren't restricted to just Longs currently -- issue when doing /users POST and trying to pass a uuid or uuid in string format. 

Only tested endpoints in /users so far and those were the issues I had run into. 

Other thing is I run docker on windows so I usually manually change the config.yml's db host to my default ip address to connect to the db. Unsure if we want to add to the config to account for different developers on different OS, but just adding this in here / also seeing if anyone has advice or suggestions on that

mbuchel commented on entrypoint.sh:
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+
+echo "*** Startup $0 suceeded now starting service using eval to expand CMD variables ***"
+exec java -jar ${JAVA_OPTS} /app.jar server /config.yml
```
inconsistent, says eval but uses exec, suggest to switch to `eval "java -jar ${JAVA_OPTS} /app.jar server /config.yml"`

toadzky replied:
we want `exec` here so that container signals are passed down to the java process.

mbuchel commented on pom.xml:
```diff
@@ -95,18 +139,42 @@
+        <!-- Database Dependencies -->
         <dependency>
-            <groupId>com.googlecode.libphonenumber</groupId>
-            <artifactId>libphonenumber</artifactId>
```
why are we no longer using libphonenumber?

toadzky replied:
it just moved down to line 175, under "Other Dependencies". no need to have a special section for single dependencies. i do the grouping so when there are multiple related dependencies (eg aws, dropwizard, gremlin), they are co-located in the file. libphonenumber doesn't have any other pieces to co-locate, so it doesn't need a separate section.

mbuchel commented on src/main/kotlin/io/zerobase/smarttracing/Main.kt:
```diff
@@ -1,28 +1,28 @@
 
-import com.google.i18n.phonenumbers.PhoneNumberUtil
 import com.fasterxml.jackson.module.kotlin.KotlinModule
+import com.google.i18n.phonenumbers.PhoneNumberUtil
```
is this not from libphonenumber?

toadzky commented:
@yonglid i think i fixed the issues with the vertexes and updated the readme accordingly. as for the different ip address thing, making it `localhost` is the most common way to use it. you should be exposing docker container ports for the database - which should make them available on localhost. that being said, if we want to leave the field as a property so it can be overridden with an environment variable in case someone can't use `localhost`, that makes sense.

toadzky committed:
scales in db cluster in stage because we don't need it

toadzky committed:
switch to https alb listener

toadzky committed:
makes local config have overridable property for db host

toadzky committed:
removes heroku files since the new build isn't compatible anyway

toadzky committed:
makes the comment match the command

toadzky committed:
fixes release workflow yaml syntax

toadzky committed:
tunes the release workflow `on` clause

toadzky committed:
don't bother building branches that don't affect the source code

toadzky committed:
adds deployment to release flow

yonglid reviewed (approved):
we can work on api routes later like you mentioned and I'm not super familiar with the deploy configurations and not familiar with branch-build yml for github actions (?); got it compiled though
